### PR TITLE
feat: config file

### PR DESCRIPTION
**Pull request by wolf31o2**
Support a YAML config file along with environment variables.

- Import `libada-go`, `yaml.v2`, and `os` in config.go
- Import `flag` in main.go
- Add `yaml` annotations for `Config` struct and children
- Set Cardano node network via `CARDANO_NETWORK` environment variable
- Default `globalConfig.App.Network` to ""
- Support `NETWORK` taking precedence over `CARDANO_NETWORK`
- Refactor `LoadConfig` to take an optional configuration file
- Create `populateNetworkMagic()` to populate `globalConfig.Node.NetworkMagic` (unused)
- Provide example configuration `config.yaml.example`
- Add license headers to source files
- Support `-config /path/to/file` to provide a YAML formatted config file
- Add `i` info to footer text for main page

image

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -44,18 +64,68 @@ var globalConfig = &Config{
...
+		case "testnet":
+			c.Node.NetworkMagic = libada.Testnet.ProtocolMagic()
+		case "mainnet":
+			c.Node.NetworkMagic = libada.Mainnet.ProtocolMagic()
+		default:
+			return fmt.Errorf("unknown network: %s", c.Node.Network)
+		}
+	}
+	return nil
+}
```
Comment (agaffney) [inline config.go:131]: FYI, there's a helper function in gOuroboros to look up a well-known network by name, and then you can retrieve the network magic value from the returned data

File: README.md [hunk 2]
```diff
@@ -25,22 +25,59 @@ Running nview against a running Cardano Node will work out of the box with a
...
 - `PROM_TIMEOUT` - Sets the maximum number of seconds to wait for response
   when polling a Cardano Node for Prometheus metrics, default is 3
+
+#### Configuration (YAML)
+
+To use a configuration file, run `nview` with a command line flag to set the
+file to load as a configuration.
+
+```bash
+./nview /path/to/config.yml
```
Comment (agaffney) [inline README.md:66]: This needs the `-config` flag

File: config.go (hunk 1 above)
Comment (wolf31o2) [reply config.go:131]: Cool. I lifted this from `tx-submit-api` directly, so I'll update both.

File: README.md (hunk 2 above)
Comment (wolf31o2) [reply README.md]: Oops! Fixed.

**Review by agaffney — APPROVED** [review]

**Pull request merged**